src/jvm/main/org/jetbrains/kotlinx/lincheck/strategy/managed/ManagedStrategy.kt

```diff
@@ -1821,6 +1870,78 @@ internal abstract class ManagedStrategy(
         enterAnalysisSection(threadId, methodSection)
     }
 
+    private fun trackAtomicMethodMemoryAccess(
```

Comment:
May I please ask you to instead make this method an extension method of `MemoryTracker` and move it to `MemoryTracker.kt` (you might need to add few more parameters to pass necessary information, e.g., `threadId`). 

`ManagedStrategy.kt` is already quite large and complicated, better to avoid enlarging it further. 

Also, I think you forgot to add a call to this method to `onMethodCall` body.